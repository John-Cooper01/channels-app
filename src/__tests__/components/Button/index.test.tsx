import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ButtonStyle from '../../../components/Button';

describe('Test button successfully', () => {
  it('Should render the button successfully', () => {
    const ButtonText = 'Fazer login';
    const { container } = render(
      <ButtonStyle variant="contained" color="primary">
        {ButtonText}
      </ButtonStyle>,
    );

    const button = screen.getByRole('button', { name: ButtonText });

    expect(button).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('Must call the button onClick function successfully', () => {
    const ButtonText = 'Fazer login';
    const onClick = jest.fn();
    render(
      <ButtonStyle variant="contained" color="primary" onClick={onClick}>
        {ButtonText}
      </ButtonStyle>,
    );

    const button = screen.getByRole('button', { name: ButtonText });

    userEvent.click(button);

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
