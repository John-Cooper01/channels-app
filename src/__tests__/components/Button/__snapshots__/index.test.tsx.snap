// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test button successfully Should render the button successfully 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Fazer login
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
